harmony/http.py

```diff
@@ -251,7 +258,34 @@ def _log_download_performance(logger, url, duration_ms, file_size):
     logger.info('timing.download.end', extra=extra_fields)
 
 
-def download(config, url: str, access_token: str, data, destination_file, user_agent=None):
+def _preprocess_download_kwargs(download_kwargs: dict, logger):
+    """Preprocess the download kwargs.
+
+    Parameters
+    ----------
+    download_kwargs: dict
+        kwargs to be passed to the download agent
+        E.g. stream=True
+    logger : logging.Logger
+        The logger to use.
+    NOTE: By default, streaming request is used to download the file,
+          and the chunksize is set to be 16MB based on the experiment with a large file of 1.8Gb
+          for optimized speed and memory consumption.
+    """
+    download_kwargs.setdefault('stream', True)
+    download_kwargs.setdefault('chunk_size', 1024*1024*16)
+    stream = download_kwargs['stream'] if 'stream' in download_kwargs else False
+    chunk_size = download_kwargs['chunk_size'] if 'chunk_size' in download_kwargs else None
+    if (not stream) and chunk_size:
+        logger.warn(
+            f"In download paramters, chunk_size={chunk_size} will be ignored since stream is set to be {stream}."
+        )
+    elif stream and not isinstance(chunk_size, int):
+        raise Exception(f"In download parameters: chunk_size must be integer when stream={stream}.")
+
+
+def download(config, url: str, access_token: str, data, destination_file,
```

Comment:
I have a few questions / comments about `**download_kwargs` here.  

1. If it stays here, it needs docs.  On other methods, you say "kwargs to be passed to the download agent", but given that this is part of the public API, it is probably worth spelling out either what the valid args are or what the agent is, so that people know how to use it.
2. Is there any case where someone wouldn't want to specify "stream=True"?  If not, do we need to parameterize it?
3. It looks like the only kwarg we actually support is `stream`, so might it be better to just provide an explicit `stream` argument?